subsys/usb/class/audio/audio.c

```diff
@@ -492,6 +564,88 @@ static struct usb_audio_dev_data *get_audio_dev_data_by_iface(uint8_t interface)
 	return NULL;
 }
 
+/**
+ * @brief Helper funciton for checking if particular endpoint is a part of
+ *	  the audio device.
+ *
+ * This function checks if given endpoint is a part of given audio device.
+ * If so then true is returned and audio_dev_data is considered correct device
+ * data.
+ *
+ * @param [in] audio_dev_data USB audio device data.
+ * @param [in] endpoint       USB Audio endpoint number.
+ *
+ * @return true if endpoint matched audio_dev_data, false otherwise.
+ */
+static bool is_endpoint_valid(struct usb_audio_dev_data *audio_dev_data,
+			      uint8_t ep, uint8_t *device)
```

Comment:
Updated the comment above the function what device is. But I agree that it would be better to handle it as a direction. 
However just having a direction would also require additional changes to DEFINE_AUDIO_DEV_DATA to know where to find what.
Since this change would change how we handle feature controls and channel counts as well it would be best to have it in a separate commit, but part of this pull request.